src/Microsoft.CodeAnalysis.Analyzers/UnitTests/MetaAnalyzers/PreferIsKindAnalyzerTests.cs

```diff
@@ -0,0 +1,423 @@
+    Microsoft.CodeAnalysis.Analyzers.MetaAnalyzers.PreferIsKindAnalyzer,
+    Microsoft.CodeAnalysis.CSharp.Analyzers.MetaAnalyzers.Fixers.CSharpPreferIsKindFix>;
+using VerifyVB = Test.Utilities.VisualBasicCodeFixVerifier<
+    Microsoft.CodeAnalysis.Analyzers.MetaAnalyzers.PreferIsKindAnalyzer,
+    Microsoft.CodeAnalysis.VisualBasic.Analyzers.MetaAnalyzers.CodeFixes.BasicPreferIsKindFix>;
+
+namespace Microsoft.CodeAnalysis.Analyzers.UnitTests.MetaAnalyzers
+{
+    public class PreferIsKindAnalyzerTests
+    {
+        [Theory]
+        [InlineData("==")]
+        [InlineData("!=")]
+        public async Task TestSimpleReturn_CS(string @operator)
+        {
+            var source =
+$@"using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+
+class C
+{{
+    bool Method(SyntaxNode node)
+    {{
+        return [|node.Kind()|] {@operator} SyntaxKind.None;
+    }}
+}}
+";
+
+            var prefix = @operator switch
+            {
+                "==" => "",
+                "!=" => "!",
+                _ => throw new InvalidOperationException(),
+            };
+
+            var fixedSource =
+$@"using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+
+class C
+{{
+    bool Method(SyntaxNode node)
+    {{
+        return {prefix}node.IsKind(SyntaxKind.None);
+    }}
+}}
+";
+
+            await VerifyCS.VerifyCodeFixAsync(source, fixedSource);
+        }
+
+        [Theory]
+        [InlineData("=")]
+        [InlineData("<>")]
+        public async Task TestSimpleReturn_VB(string @operator)
+        {
+            var source =
+$@"Imports Microsoft.CodeAnalysis
+Imports Microsoft.CodeAnalysis.VisualBasic
+
+Class C
+    Function Method(node As SyntaxNode) As Boolean
+        Return [|node.Kind()|] {@operator} SyntaxKind.None
+    End Function
+End Class
+";
+
+            var prefix = @operator switch
+            {
+                "=" => "",
+                "<>" => "Not ",
+                _ => throw new InvalidOperationException(),
+            };
+
+            var fixedSource =
+$@"Imports Microsoft.CodeAnalysis
+Imports Microsoft.CodeAnalysis.VisualBasic
+
+Class C
+    Function Method(node As SyntaxNode) As Boolean
+        Return {prefix}node.IsKind(SyntaxKind.None)
+    End Function
+End Class
+";
+
+            await VerifyVB.VerifyCodeFixAsync(source, fixedSource);
+        }
+
+        [Fact]
+        public async Task TestCompoundExpression_CS()
+        {
+            var source =
+@"using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+
+class C
+{
+    bool Method(SyntaxNode node)
+    {
+        return [|node.Kind()|] != SyntaxKind.None && [|node.Kind()|] != SyntaxKind.TrueKeyword;
+    }
+}
+";
+            var fixedSource =
+@"using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+
+class C
+{
+    bool Method(SyntaxNode node)
+    {
+        return !node.IsKind(SyntaxKind.None) && !node.IsKind(SyntaxKind.TrueKeyword);
```

Comment:
If there are overloads that take multiple kinds, it would look nicer to use them. But that's not a necessary change to do, especially if it will increase the complexity and decrease performance due its limited value.

I'd do that only if it's easy and cheap enough.

Reply:
This analyzer is implemented for all Roslyn users, but those methods are not public APIs so I didn't include them in the analysis.